api/environments/tests/test_models.py

```diff
@@ -55,6 +55,30 @@ def test_on_creation_save_feature_is_created_with_the_correct_default(self):
         self.environment.save()
         self.assertFalse(FeatureState.objects.get().enabled)
 
+    def test_clone_does_not_modify_the_original_instance(self):
+        # Given
+        self.environment.save()
+
+        # When
+        clone = self.environment.clone(name="Cloned env")
+        clone.save()
+
+        # Then
+        self.assertNotEqual(clone.name, self.environment.name)
+        self.assertNotEqual(clone.api_key, self.environment.api_key)
+
+    def test_clone_save_creates_feature_states(self):
+        # Given
+        self.environment.save()
+
+        # When
+        clone = self.environment.clone(name="Cloned env")
+        clone.save()
+
+        # Then
+        feature_states = FeatureState.objects.filter(environment=clone)
+        assert feature_states.count() == 1
```

Comment:
I wonder if we should check that the feature states of the source environment haven't been modified too?